src/HazelcastClient.ts

```diff
@@ -7,39 +7,42 @@ import Q = require('q');
 import {IMap} from './IMap';
 import {JsonSerializationService} from './serialization/SerializationService';
 import PartitionService = require('./PartitionService');
+import ClusterService = require('./invocation/ClusterService');
 
 class HazelcastClient {
-    private config: ClientConfig;
+
+    private config: ClientConfig = new ClientConfig();
     private serializationService: SerializationService;
     private invocationService: InvocationService;
     private connectionManager: ClientConnectionManager;
     private partitionService: PartitionService;
+    private clusterService: ClusterService;
     private proxyManager: ProxyManager;
 
     public static newHazelcastClient(config?: ClientConfig): Q.Promise<HazelcastClient> {
         var client: HazelcastClient = new HazelcastClient(config);
-        var clientPromise: Q.Promise<HazelcastClient> = client.init();
-        return clientPromise;
+        return client.init();
     }
 
     constructor(config?: ClientConfig) {
-        if (!config) {
-            this.config = new ClientConfig();
```

Comment:
What if client does not provide a config? How do we provide default configuration?